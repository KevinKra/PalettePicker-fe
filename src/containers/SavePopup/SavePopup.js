import React, { Fragment } from "react";
import { connect } from "react-redux";
import * as actions from "../../_redux/actions";
import "./SavePopup.scss";

function SavePopup(props) {
  const handleExit = () => {
    props.history.push("/");
  };

  const removeColor = color => {
    const paletteCopy = props.currentPalette.slice();
    const colorIndex = props.currentPalette.findIndex(
      paletteColor => paletteColor.hex === color.hex
    );
    paletteCopy[colorIndex].locked = false;
    props.updateCurrentPalette(paletteCopy);
  };

  const colorOptions = props.currentPalette
    .filter(palette => palette.locked === true)
    .map(palette => {
      return (
        <article className="palette-details" key={palette.hex}>
          <section className="palette-inputs">
            <div
              className="palette-color"
              style={{
                backgroundColor: palette.hex
              }}
            ></div>
            <input type="text" defaultValue={palette.hex} />
          </section>
          <button
            className="remove-btn"
            type="button"
            onClick={() => removeColor(palette)}
          >
            Remove
          </button>
        </article>
      );
    });
  return (
    <Fragment>
      <div className="screen" />
      <section className="SavePopup">
        <button className="editor-exit" onClick={handleExit}>
          X
        </button>
        <form className="editor-form">
          <h3 className="editor-title">Title</h3>
          <label htmlFor="">Label 1</label>
          <input type="text" />
          <section className="palettes-section">{colorOptions}</section>
          <button className="submit-btn">Submit</button>
        </form>
      </section>
    </Fragment>
  );
}

const mapStateToProps = store => ({
  currentPalette: store.currentPalette
});

const mapDispatchToProps = dispatch => ({
  updateCurrentPalette: palette =>
    dispatch(actions.updateCurrentPalette(palette))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SavePopup);
