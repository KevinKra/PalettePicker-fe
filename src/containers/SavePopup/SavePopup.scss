@import "../../_styles/variables.scss";

.SavePopup-container {
  position: fixed;
  margin-left: auto;
  margin-right: auto;
  top: 20%;
  left: 15%;
  right: 15%;
  z-index: 4;
  width: 400px;
}

.save-section {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  position: absolute;
  z-index: 1;
  bottom: 0;
  left: 0;
  border: 1px solid $primary-color;
  border-bottom: 5px solid $primary-color;
  width: 100%;
  height: 155px;
  transition: 0.8s all cubic-bezier(0.13, 1.28, 0.5, 1.14);
  background-color: $primary-light;
  .submit-btn {
    background-color: $primary-color;
    color: $primary-light;
    padding: 0.7rem;
    margin-bottom: 0.3rem;
    border: 1px solid $primary-light;
    font-size: 14px;
    width: 98%;
  }
  .query-inputs > section {
    // border: 1px solid red;
    display: flex;
    flex-direction: column;
    padding-top: 0.5rem;
  }

  .query-param > input {
    margin: 0.2rem 0;
    @include input-focus;
  }

  h3 {
    margin: 0;
    margin-bottom: 0.2rem;
  }
}

.SavePopup {
  position: relative;
  z-index: 2;
  width: 100%;
  border: 1px solid $primary-color;
  border-bottom: 5px solid $primary-color;
  background-color: $primary-light;
  transition: 0.7s height ease-in-out;

  .title-background {
    position: absolute;
    top: -45px;
    left: -15px;
    transform: skew(15deg);
    padding: 0.7rem;
    background-color: $primary-color;
  }

  .editor-title {
    transform: skew(-15deg);
    color: $primary-light;
    font-family: $font-3;
  }

  .exit-background {
    position: absolute;
    right: -15px;
    top: -15px;
    padding: 0.4rem;
    padding-top: 0.7rem;
    transform: skew(-12deg);
    background-color: $primary-color;
  }

  .save-exit {
    background-color: $primary-dark;
    color: $primary-light;
    border: 1px solid $primary-dark;
    transform: skew(-4deg);
    font-size: 15px;
    padding: 0.3rem 0.5rem;
  }

  .palette-form {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0.6rem 0.5rem;
  }

  .palette-form > header {
    display: flex;
    align-items: center;
    padding: 1rem;
  }

  .palette-form > header > label {
    font-size: 16px;
    text-decoration: underline;
  }

  .palette-form > header > input {
    font-size: 14px;
    padding: 0.1rem;
    margin-left: 0.3rem;
    @include input-focus;
  }

  .palettes-section {
    background-color: $primary-light;
    border-right: 0.5rem solid white;
    border-left: 0.5rem solid white;
    width: 90%;
    max-height: 20vh;
    overflow: scroll;
  }

  button {
    border: 1px solid $primary-color;
    background-color: $primary-color;
    color: $primary-light;
    padding: 0.7rem;
    margin: 0.2rem 0;
    font-size: 14px;
    width: 100%;
  }
}
