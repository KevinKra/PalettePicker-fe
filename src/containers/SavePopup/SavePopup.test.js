import React from "react";
import { shallow } from "enzyme";
import { SavePopup } from "./SavePopup";

describe("SavePopup", () => {
  it("matches snapshot", () => {
    const props = {
      currentPalette: [
        {
          locked: true
        }
      ],
      userProjects: [{ projectId: 8 }]
    };
    const wrapper = shallow(<SavePopup {...props} />, {
      disableLifecycleMethods: true
    });
    expect(wrapper).toMatchSnapshot();
  });
});
