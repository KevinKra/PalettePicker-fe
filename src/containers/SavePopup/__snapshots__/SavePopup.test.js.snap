// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavePopup matches snapshot 1`] = `
<Fragment>
  <div
    className="screen"
  />
  <div
    className="SavePopup-container"
  >
    <section
      className="SavePopup"
    >
      <div
        className="exit-background"
      >
        <button
          className="save-exit"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="title-background"
      >
        <h3
          className="editor-title"
        >
          Save Your Palette
        </h3>
      </div>
      <form
        className="palette-form"
      >
        <header>
          <p
            className="error-message"
          />
          <label
            htmlFor="palette-title"
          >
            Palette Title:
          </label>
          <input
            autoComplete="off"
            maxLength="25"
            name="paletteTitle"
            onChange={[Function]}
            type="text"
          />
        </header>
        <section
          className="palettes-section"
        >
          <ColorEditor
            handleColor={[Function]}
            key="NaN"
            palette={
              Object {
                "locked": true,
              }
            }
          />
        </section>
        <div
          className="query-buttons"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Save to an Existing Project
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Create New Project
          </button>
        </div>
      </form>
    </section>
    <section
      className="save-section"
      style={
        Object {
          "transform": "translateY(0%)",
        }
      }
    >
      <div
        className="query-inputs"
      />
    </section>
  </div>
</Fragment>
`;
